int32 allied_type
---
float32 gamma
int32 error